Modules/Sources/Experiments/DefaultFeatureFlagService.swift

```diff
@@ -100,6 +100,8 @@ public struct DefaultFeatureFlagService: FeatureFlagService {
             return buildConfig == .localDeveloper || buildConfig == .alpha
         case .ciabBookings:
             return buildConfig == .localDeveloper || buildConfig == .alpha
+        case .liquidGlassDesign:
+            return buildConfig == .localDeveloper || buildConfig == .alpha
```

Comment:
I suggest having something a bit different logic from a normal feature flag.

```suggestion
            let requiresCompatibility = Bundle.main.infoDictionary?["UIDesignRequiresCompatibility"] as? Bool ?? false
            let isIOS26OrHigher = ProcessInfo.processInfo.operatingSystemVersion.majorVersion >= 26
            return !requiresCompatibility && isIOS26OrHigher
```

Since we already have a flag within the Info.plist that will enable `Liquid Glass`, I think at that point we will want to have our changes such as this enabled as well, otherwise the app could crash in production if we keep it as `buildConfig == .localDeveloper || buildConfig == .alpha`. 

With the updated logic, our liquid glass design changes will only be enabled if:
- `UIDesignRequiresCompatibility` flag doesn't exist or is set on `NO`
- User is on iOS 26 or higher version. We don't want our new badges to be shown to users who are on iOS 18.5 even if we enable liquid glass for iOS 26 users. 

Wdyt?

Reply:
Thanks for the suggestion. Originally, I used a local feature flag so that it's more obvious that the code can be removed once the Liquid Glass design is launched. However, it does rely on a strict sync with the removal of the Info.plist entry, which cannot be guaranteed if some miscommunication happens. I made an update in https://github.com/woocommerce/woocommerce-ios/pull/16161/commits/d968d88dd2aa2ba3df02d62f6646826d0ebb8226 to replace the feature flag check with the Info.plist check as you suggested.

Regarding the iOS 26+ check, I'm leaving it out so that the native tab badge is applied to all iOS versions once the Liquid Glass design is enabled. iOS version adoption is usually pretty fast, and I don't think showing the native tab badge design causes critical UI/UX issues. We can remove the legacy fragile code sooner.